app/src/ui/app.tsx

```diff
@@ -488,12 +471,42 @@ export class App extends React.Component<IAppProps, IAppState> {
       return
     }
 
+    this.props.dispatcher.closeCurrentFoldout()
     this.props.dispatcher.changeRepositorySection(
       state.repository,
       RepositorySection.History
     )
   }
 
+  private chooseRepository() {
+    if (
+      this.state.currentFoldout &&
+      this.state.currentFoldout.type === FoldoutType.Repository
+    ) {
+      return this.props.dispatcher.closeFoldout(FoldoutType.Repository)
+    }
+
+    return this.props.dispatcher.showFoldout({
+      type: FoldoutType.Repository,
+    })
+  }
+
+  private showBranches() {
+    const state = this.state.selectedState
+    if (!state || state.type !== SelectionType.Repository) {
```

Comment:
`state != null`

Reply:
Addressed in 08a435d9b439193d70663afb591c42ef145100e8. 🕵️